helm/system-upgrade-contoller/templates/plan.yaml

```diff
@@ -0,0 +1,44 @@
+{{- if .Values.installPlan -}}
+# Server plan
+apiVersion: upgrade.cattle.io/v1
+kind: Plan
+metadata:
+  name: server-plan
+  namespace: {{.Values.targetns }}
+spec:
+  concurrency: 1
+  cordon: true
+  nodeSelector:
+    matchExpressions:
+    - key: node-role.kubernetes.io/control-plane
+      operator: In
+      values:
+      - "true"
+  serviceAccountName: system-upgrade
+  upgrade:
+    image: rancher/k3s-upgrade
+  version: v1.24.6+k3s1
```

Comment:
i don't know if this is exactly relevant, but this should match the version of kubectl you're deploying with.

or at least, it should be an input in values.yaml